sdk/storage/storage-file/review/storage-file.api.ts

```diff
@@ -0,0 +1,787 @@
+  setMetadata(metadata?: Metadata, options?: DirectorySetMetadataOptions): Promise<Models.DirectorySetMetadataResponse>;
+  setProperties(properties?: DirectoryProperties): Promise<Models.DirectorySetPropertiesResponse>;
+  // (undocumented)
+  readonly shareName: string;
+}
+
+// @public
+interface DirectoryCreateOptions extends FileAndDirectoryCreateCommonOptions, CommonOptions {
+  abortSignal?: AbortSignalLike;
+  metadata?: Metadata;
+}
+
+// @public
+interface DirectoryDeleteOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface DirectoryForceCloseHandlesOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface DirectoryForceCloseHandlesSegmentOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+  recursive?: boolean;
+}
+
+// @public
+interface DirectoryGetPropertiesOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface DirectoryListFilesAndDirectoriesOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+  prefix?: string;
+}
+
+// @public
+interface DirectoryListHandlesOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+  recursive?: boolean;
+}
+
+// @public
+interface DirectoryListHandlesSegmentOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+  maxresults?: number;
+  recursive?: boolean;
+}
+
+// @public (undocumented)
+interface DirectoryProperties extends FileAndDirectorySetPropertiesCommonOptions, CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface DirectorySetMetadataOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface DownloadFromAzureFileOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+  maxRetryRequestsPerRange?: number;
+  parallelism?: number;
+  progress?: (progress: TransferProgressEvent) => void;
+  rangeSize?: number;
+}
+
+// @public
+interface FileAbortCopyFromURLOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+interface FileClearRangeOptions extends CommonOptions {
+  abortSignal?: AbortSignalLike;
+}
+
+// @public
+class FileClient extends StorageClient {
+  constructor(url: string, credential?: Credential, options?: NewPipelineOptions);
+  abortCopyFromURL(copyId: string, options?: FileAbortCopyFromURLOptions): Promise<Models.FileAbortCopyResponse>;
+  clearRange(offset: number, contentLength: number, options?: FileClearRangeOptions): Promise<Models.FileUploadRangeResponse>;
+  create(size: number, options?: FileCreateOptions): Promise<Models.FileCreateResponse>;
+  delete(options?: FileDeleteOptions): Promise<Models.FileDeleteResponse>;
+  download(offset?: number, count?: number, options?: FileDownloadOptions): Promise<Models.FileDownloadResponse>;
+  downloadToBuffer(buffer: Buffer, offset: number, count?: number, options?: DownloadFromAzureFileOptions): Promise<void>;
```

Comment:
>DownloadFromAzureFileOptions [](start = 77, length = 28)

`DownloadFromAzureFileOptions` not align with `FileDownloadOptions`.

Reply:
Which field(s)?  I believe one doesn't support concurrency thus the diff. in the option types.

Reply:
Could use same `FileDownloadOptions` if we could ignore concurrency etc.

Reply:
Yes, just about the naming of option models. "download()" => "FileDownloadOptions", while "downloadToBuffer" => "DownloadFromAzureFileOptions" Naming pattern not aligned

Reply:
#6022